crates/base-db/src/lib.rs

```diff
@@ -43,6 +44,26 @@ macro_rules! impl_intern_key {
     };
 }
 
+#[macro_export]
+macro_rules! impl_intern_key_ref {
+    ($id:ident, $loc:ident) => {
+        #[salsa::interned(no_debug, no_lifetime)]
+        pub struct $id {
+            #[return_ref]
```

Comment:
Next salsa will hopefully flip this to be the default mode (once someone implements it https://github.com/salsa-rs/salsa/issues/719)